app/src/main/java/com/duckduckgo/app/appearance/AppearanceActivity.kt

```
@@ -200,8 +200,9 @@ class AppearanceActivity : DuckDuckGoActivity() {
         val subtitle =
             getString(
                 when (omnibarType) {
-                    OmnibarType.SINGLE_TOP, OmnibarType.SPLIT -> R.string.settingsAddressBarPositionTop
+                    OmnibarType.SINGLE_TOP -> R.string.settingsAddressBarPositionTop
                     OmnibarType.SINGLE_BOTTOM -> R.string.settingsAddressBarPositionBottom
+                    OmnibarType.SPLIT -> throw IllegalStateException("Split omnibar not supported in the position selection settings")
```

do you really wanna throw here? if this is some sort of state we don’t want to have, maybe it’s better to prevent it instead. Just crashing without doing or telling anything to the user is a bit drastic.

---

This isn't meant to crash an app for a user, but rather let us find a bug early when testing (basically the [fail fast principle](https://en.wikipedia.org/wiki/Fail-fast_system)). Also, once we remove the feature flags, we'll get rid of it.

This code is used to display the old settings where there's nothing to show for the `SPLIT` omnibar type. It's executed only when the required feature flags are disabled, meaning [OmnibarFeatureRepository.isSplitOmnibarAvailable is false](https://github.com/duckduckgo/Android/blob/90de2302f634b1cf7d4caa9393fe39699d7996a1/app/src/main/java/com/duckduckgo/app/appearance/AppearanceViewModel.kt#L96). We should never get here but we still need to "handle" the new omnibar type somehow. If we do get here, it means we have bigger problems/inconsistencies because the flags are used all over the place. There already is a [mechanism to prevent this state](https://github.com/duckduckgo/Android/blob/90de2302f634b1cf7d4caa9393fe39699d7996a1/app/src/main/java/com/duckduckgo/app/browser/omnibar/OmnibarFeatureRepository.kt#L68).